src/main/webapp/js/datahandlers/scholarshipdatahandler.js

```diff
@@ -59,20 +60,22 @@ class ScholarshipDataHandler {
     }
 
```

Comment:
What is this `key` data used for?

I'm really confused about what the requirementsMap is doing. REQUIREMENTS isn't a map so there will never be a value. Also you're only iterating through fields in REQUIREMENTS so you'll never hit the else block.